libxrdp/xrdp_sec.c

```diff
@@ -2298,159 +2298,190 @@ xrdp_sec_process_mcs_data_channels(struct xrdp_sec *self, struct stream *s)
-        in_uint32_le(s, client_info->minfo[index].top);
-        in_uint32_le(s, client_info->minfo[index].right);
-        in_uint32_le(s, client_info->minfo[index].bottom);
-        in_uint32_le(s, client_info->minfo[index].is_primary);
-
-        LOG_DEVEL(LOG_LEVEL_TRACE, "Received [MS-RDPBCGR] "
-                  "TS_UD_CS_MONITOR.TS_MONITOR_DEF %d "
-                  "left %d, top %d, right %d, bottom %d, flags 0x%8.8x",
-                  index,
-                  client_info->minfo[index].left,
-                  client_info->minfo[index].top,
-                  client_info->minfo[index].right,
-                  client_info->minfo[index].bottom,
-                  client_info->minfo[index].is_primary);
-
-        if (index == 0)
-        {
-            x1 = client_info->minfo[index].left;
-            y1 = client_info->minfo[index].top;
-            x2 = client_info->minfo[index].right;
-            y2 = client_info->minfo[index].bottom;
+
+        monitor_layout = description->minfo + monitor_index;
+        if (full_parameters != 0) {
+            in_uint32_le(s, monitor_layout->flags);
+        }
+        in_uint32_le(s, monitor_layout->left);
+        in_uint32_le(s, monitor_layout->top);
+        in_uint32_le(s, monitor_layout->width);
+        in_uint32_le(s, monitor_layout->height);
+        if (full_parameters != 0) {
+            in_uint32_le(s, monitor_layout->physical_width);
+            in_uint32_le(s, monitor_layout->physical_height);
+            in_uint32_le(s, monitor_layout->orientation);
+            in_uint32_le(s, monitor_layout->desktop_scale_factor);
+            in_uint32_le(s, monitor_layout->device_scale_factor);
+        }
+
+        monitor_layout->right = monitor_layout->left + monitor_layout->width;
+        monitor_layout->bottom = monitor_layout->top + monitor_layout->height;
+
+        if (full_parameters != 0)
+        {
+            LOG(LOG_LEVEL_INFO, "    Index: %d, Flags 0x%8.8x Left %d Top %d "
+                "Width %d Height %d PhysicalWidth %d PhysicalHeight %d "
+                "Orientation %d DesktopScaleFactor %d DeviceScaleFactor %d",
+                monitor_index, monitor_layout->flags, monitor_layout->left,
+                monitor_layout->top, monitor_layout->width, monitor_layout->height,
+                monitor_layout->physical_width, monitor_layout->physical_height,
+                monitor_layout->orientation, monitor_layout->desktop_scale_factor,
+                monitor_layout->device_scale_factor);
         }
         else
         {
-            x1 = MIN(x1, client_info->minfo[index].left);
-            y1 = MIN(y1, client_info->minfo[index].top);
-            x2 = MAX(x2, client_info->minfo[index].right);
-            y2 = MAX(y2, client_info->minfo[index].bottom);
+            LOG_DEVEL(LOG_LEVEL_TRACE, "Received [MS-RDPBCGR] "
+                "TS_UD_CS_MONITOR.TS_MONITOR_DEF %d "
+                "left %d, top %d, right %d, bottom %d, flags 0x%8.8x",
```

Comment:
why are these two log messages at different levels and using different formats? 
can you please make them consistent, and include the spec and struct reference in the messages.

Reply:
Updated.